### PR TITLE
[INLONG-4262][Sort] Add SqlServer data extract support

### LINKED ISSUE #4262: [Feature][Sort] Add SqlServer data extract support
### Description

Add SqlServer data extract support

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by GanfengTan**
Add SqlServer data extract support

Fixes #4262

### CODE REVIEW

**Review by healchow — APPROVED** [review]

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/SqlServerExtractNode.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+@EqualsAndHashCode(callSuper = true)
+@JsonTypeName("sqlserverExtract")
+@Data
+public class SqlServerExtractNode extends ExtractNode implements Serializable {
+
+    private static final long serialVersionUID = 5096171152872086083L;
+
+    @JsonProperty("hostname")
+    @Nonnull
+    private String hostname;
```
Comment (healchow) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/SqlServerExtractNode.java:48]: These variables are too crowded and less readable, can you add a blank line between them?

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/SqlServerExtractNode.java (hunk 1 above)
Comment (GanfengTan) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/SqlServerExtractNode.java:48]: very good advice

**Review by gong — APPROVED** [review]

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/SqlServerExtractNode.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import java.util.List;
+import java.util.Map;
+
+/**
+ * SqlServer extract node using debezium engine.
+ */
+@EqualsAndHashCode(callSuper = true)
+@JsonTypeName("sqlserverExtract")
+@Data
+public class SqlServerExtractNode extends ExtractNode implements Serializable {
```
Comment (EMsnap) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/SqlServerExtractNode.java:42]: please add a test for serialization and deserialization

Comment (GanfengTan) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/SqlServerExtractNode.java:42]: done

**Review by EMsnap — APPROVED** [review]

**Pull request merged**